### DISCUSSION TITLE
Selecting multiple points on scatterplot layer

### DISCUSSION THREAD

**Discussion by btbritz** [visgl/deck.gl#7072] [Q&A] [answered]
Is there a way to to select multiple points on a scatterplot layer? Maybe by using a lasso, or some other selection tool, to click and drag and through an entire area on the layer and get returned the point data?

Thanks

**Comment by mcost45** [comment, marked as answer]
https://deck.gl/docs/api-reference/core/deck#pickobjects

But you will need to implement the visualisation/user input/event handling to pass in a bounding box

**Reply by btbritz** [reply]
Awesome! Thanks for the response.

There wouldn't happen to be an example out there that can step you through that implementation would there?

Thanks again

**Reply by mcost45** [reply]
Hey sorry, not that I know of 😅 Though it would be along these lines:

1. Add a new path layer for visualising the selection box, you can use the PathStyleExtension to make it dashed etc
2. Set up event callbacks for dragging via deck props
3. If the lasso tool is selected, disable the views controller so that dragging won't move the screen. Enable controller again when lasso tool is unselected. Clear/Toggle visibility of path layer when lasso tool is selected/unselected 
4. On drag start you'll get the starting pos
5. On drag you'll get the end pos, using both positions update the data of the path layer to visualise the current selection box. You can also use this box to pick everything that is contained
6. On drag end the selection is completed

**Reply by btbritz** [reply]
Awesome! Thanks for the quick response, this helped a bunch!
